src/notifications/useMarkNotificationAsSeen.ts

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useState } from 'react';
+
+import { IMessageId } from '@novu/headless';
+
+import { useHeadlessService } from './useHeadlessService';
+
+const useMarkNotificationAsSeen = () => {
+  const { headlessService, isReady } = useHeadlessService();
+  const [isMutating, setIsMutating] = useState(false);
+  const [error, setError] = useState<unknown>(null);
+
+  const markNotificationsAsSeen = useCallback(
+    (messageIds: IMessageId) => {
+      if (isReady) {
+        headlessService.markNotificationsAsSeen({
+          messageId: messageIds,
+          listener: ({ isLoading, isError, error: err }) => {
+            setIsMutating(isLoading);
+
+            if (isError) {
```

Comment:
Report error to Sentry either from here or from inside `onError` which is what I prefer